tests/test_basic.py

```diff
@@ -6,50 +6,54 @@
 
 def test_happy(monkeypatch):
     mock_publisher = Mock()
-    monkeypatch.setattr(
-        "google.cloud.pubsub_v1.PublisherClient.publish", mock_publisher
-    )
+    monkeypatch.setattr("google.cloud.pubsub_v1.PublisherClient.publish", mock_publisher)
 
-    tap_ps = subprocess.Popen(
-        ("cat", "./tests/dummy_messages.txt"), stdout=subprocess.PIPE
-    )
+    tap_ps = subprocess.Popen(("cat", "./tests/dummy_messages.txt"), stdout=subprocess.PIPE)
 
     tap_ps.wait()
     state = main(tap_ps.stdout)
-    assert mock_publisher.call_count == 5  # one for each line in dummy_messages.txt
+    assert mock_publisher.call_count == 3  # one for each record in dummy_messages.txt
 
     calls = [
         call(
             "projects/xyz/topics/users",
-            data=b'{"type": "SCHEMA", "stream": "users", "key_properties": ["id"], "schema": {"required": ["id"], "type": "object", "properties": {"id": {"type": "integer"}}}}',
+            data=b'{"type": "RECORD", "stream": "users", "record": {"id": 1, "name": "Chris"}, "schema": {"required": ["id"], "type": "object", "properties": {"id": {"type": "integer"}}}}',  # noqa: E501
             stream="users",
         ),
         call().result(),
         call(
             "projects/xyz/topics/users",
-            data=b'{"type": "RECORD", "stream": "users", "record": {"id": 1, "name": "Chris"}}',
+            data=b'{"type": "RECORD", "stream": "users", "record": {"id": 2, "name": "Mike"}, "schema": {"required": ["id"], "type": "object", "properties": {"id": {"type": "integer"}}}}',  # noqa: E501
             stream="users",
         ),
         call().result(),
-        call(
-            "projects/xyz/topics/users",
-            data=b'{"type": "RECORD", "stream": "users", "record": {"id": 2, "name": "Mike"}}',
-            stream="users",
-        ),
-        call().result(),
-        call(
-            "projects/xyz/topics/locations",
-            data=b'{"type": "SCHEMA", "stream": "locations", "key_properties": ["id"], "schema": {"required": ["id"], "type": "object", "properties": {"id": {"type": "integer"}}}}',
-            stream="locations",
-        ),
-        call().result(),
         call(
             "projects/xyz/topics/locations",
-            data=b'{"type": "RECORD", "stream": "locations", "record": {"id": 1, "name": "Philadelphia"}}',
+            data=b'{"type": "RECORD", "stream": "locations", "record": {"id": 1, "name": "Philadelphia"}, "schema": {"required": ["id"], "type": "object", "properties": {"id": {"type": "integer"}}}}',  # noqa: E501
             stream="locations",
         ),
         call().result(),
     ]
 
     mock_publisher.assert_has_calls(calls)
-    assert state == {"users": 2, "locations": 1}
+    assert state == {
```

Comment:
The state still needs to conform to the singer spec as it'll need to get read by the postgres-tap on the next run right?

Reply:
The [spec didn't say anything about it](https://github.com/singer-io/getting-started/blob/master/docs/SPEC.md#state-message) so I thought I'll just improvise something 🤷‍♂ Wait, just realised, so the value should be the last record ID (or whatever is the primary key) for each stream?